Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoMeraki.yaml

```diff
@@ -1,46 +1,580 @@
+            Syslog
+            | project-rename LogMessage =  SyslogMessage
+            )
+        | where not(disabled) and (LogMessage has_any("flows", "firewall", "ids-alerts") or LogMessage has_all("security_event", "ids-alerted") or (LogMessage has "events" and (LogMessage has_any ("Blocked DHCP server response", "association") or (LogMessage has "VRRP packet" and not(LogMessage has_any ("VRRP passive", "VRRP active"))) or (LogMessage has "disassociation" and LogMessage !has "auth_neg_failed"))) or (LogMessage has "airmarshal_events" and LogMessage has_any("ssid_spoofing_detected", "rogue_ssid_detected")))
+        | extend Parser = extract_all(@"(\d+.\d+)\s([\w\-\_]+)\s([\w\-\_]+)\s([\S\s]+)$", dynamic([1, 2, 3, 4]), LogMessage)[0]
+        | extend
+            LogType = tostring(Parser[2]),
+            Substring = tostring(Parser[3])
+        | extend LogSubType = case( 
+                            LogType == "security_event",
+                            tostring(extract(@"^(\S+)\s", 1, Substring)),
+                            (LogType == "events" and Substring has "Blocked DHCP server response"),
+                            "Blocked DHCP server response",
+                            LogType == "events" and (Substring has "VRRP packet" and not(Substring in~ ("VRRP passive", "VRRP active"))),
+                            "Virtual router collision",
+                            LogType in~ ("events", "airmarshal_events"),
+                            extract(@"type[ ]{0,1}[=]{0,1}[ ]{0,1}(\S+)", 1, Substring),
+                            ""
+                        )
+        | where (LogType in("flows", "firewall", "ids-alerts") or (LogType == "security_event" and LogSubType == "ids-alerted") or (LogType == "events" and (LogSubType in ("Blocked DHCP server response", "association", "Virtual router collision") or (LogSubType has "disassociation" and LogSubType !has "auth_neg_failed"))) or (LogType == "airmarshal_events" and LogSubType in("ssid_spoofing_detected", "rogue_ssid_detected")))
+        | parse-kv Substring as(timestamp: string) with (pair_delimiter=" ", kv_delimiter="=")
+        | extend 
+            Epoch = tostring(Parser[0]),
+            Device = tostring(Parser[1])
+        | extend
+            Epoch = iff(isnotempty(column_ifexists("timestamp", "")), timestamp, Epoch)
+        | extend
+            EpochTimestamp = split(Epoch, ".")
+        | extend
+            EventStartTime = unixtime_seconds_todatetime(tolong(EpochTimestamp[0]))
+        | parse-kv Substring as(src: string, ip_src: string, last_known_client_ip: string, ip: string, client_ip: string, dst: string, dns_server: string, client_mac: string, mac: string, shost: string, dhost: string, wired_mac: string, identity: string, sport: string, dport: string, direction: string, reason: string, protocol: string, priority: string, duration: string, vlan_id: string, signature: string, aid: string, rssi: string) with (pair_delimiter=" ", kv_delimiter="=")
```

Comment:
Also, some of the values might contain quotes and thus the better condition would be to use the following:
with (pair_delimiter=' ', kv_delimiter='=', quote="'")

Reply:
Done